Sources/StreamChat/WebSocketClient/ConnectionStatus.swift

```diff
@@ -42,9 +42,7 @@ extension ConnectionStatus {
             self = .disconnecting
 
         case let .disconnected(source):
-            let isWaitingForReconnect = webSocketConnectionState.isAutomaticReconnectionEnabled || source.serverError?
-                .isInvalidTokenError == true
```

Comment:
why do we remove the invalid token condition?

Reply:
Because it does not make sense, if the token is invalid, it should not mean we are reconnecting, it should stop trying to automatically reconnect. In fact, inside `isAutomaticReconnectionEnabled` this error makes it `false`. But then here we set it to `true`

Reply:
the var name is `isWaitingForReconnect` though - it determines whether the state should be connecting or disconnected. If it's an invalid token error, doesn't that mean that maybe a token refresh is in progress?

Reply:
Inside the `isAutomaticReconnectionEnabled` we have this:

<img width="581" alt="image" src="https://github.com/user-attachments/assets/399cddc2-9075-494d-8385-f1b5945116fa">

The problem, it seems, is that we have this:

<img width="934" alt="image" src="https://github.com/user-attachments/assets/33c1677e-926e-44f3-a6f2-c3dcd3d0f8d1">

And it looks like this includes expired tokens:

<img width="1057" alt="image" src="https://github.com/user-attachments/assets/d305220f-1e2b-4e51-b7e4-556c4f732cf4">

So I guess this should be updated

Reply:
yes, this doesn't make much sense - but this was added explicitly 2 years ago it seems, so we should be careful with this change.

Reply:
Done ✅ Ive made some changes and I think now it makes much more sense